.github/workflows/build_release.yml

```diff
@@ -58,32 +58,23 @@ jobs:
       run: find bin/ -type f -perm -u=x -exec strip {} +
       if: matrix.os != 'windows-latest'
 
-    - name: archive
-      id: archive
-      run: |
-        OSNAME=$(echo ${{ matrix.os }} | sed 's/-latest//')
-        VERSION=${{ github.event.release.tag_name }}
-        PKGNAME="wabt-$VERSION-$OSNAME"
-        TARBALL=$PKGNAME.tar.gz
-        SHASUM=$PKGNAME.tar.gz.sha256
-        mv install wabt-$VERSION
-        tar -czf $TARBALL wabt-$VERSION
-        scripts/sha256sum.py $TARBALL > $SHASUM
-        echo "::set-output name=tarball::$TARBALL"
-        echo "::set-output name=shasum::$SHASUM"
+    - uses: ./.github/actions/upload-release-archive
+      with:
+        os: ${{ matrix.os }}
 
-    - name: upload tarball
-      uses: svenstaro/upload-release-action@v2
+  build-wasi:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
       with:
-        repo_token: ${{ secrets.GITHUB_TOKEN }}
-        file: ./${{ steps.archive.outputs.tarball }}
-        asset_name: ${{ steps.archive.outputs.tarball }}
-        tag: ${{ github.ref }}
+        submodules: true
+
+    - name: built-tools
+      run: |
+        docker run -di --name wasi-sdk -v $(pwd):/src --workdir /src ghcr.io/webassembly/wasi-sdk:wasi-sdk-20 bash
+        docker exec wasi-sdk cmake -S . -B out -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=install -DBUILD_TESTS=0 -DBUILD_LIBWASM=0
+        docker exec wasi-sdk cmake --build out --config Release --target install
 
-    - name: upload shasum
-      uses: svenstaro/upload-release-action@v2
+    - uses: ./.github/actions/upload-release-archive
```

Comment:
I seem to remember there was a simple way to have a build action upload an artifact during CI (different to adding release artifacts).     Might be cool to enable that so we can download and try out the binaries that get generated by this PR?

Otherwise LGTM!

Reply:
Yeah it's pretty simple using the upload-artifact step

https://github.com/actions/upload-artifact

In that case I would add the step here, guard the release upload steps to only happen on a release, and use this action from `build.yml`.

I can try the change but I would probably only have it added to wasi then for now - the other steps in `build.yml` don't actually generate a release archive. I think in general it is a good idea to do so as it reduces the chance of CI passing + release failing, so I took the approach with wasi, but I don't want to get too invasive on non-wasi parts of the workflow in this PR if it's OK. So should we just add artifact uploading in `build.yml` for wasi?

Reply:
sgtm